dali/operators/random/rng_base_gpu.cuh

```diff
@@ -75,39 +126,64 @@ __global__ void RNGKernel(BlockDesc<IsNoiseGen>* __restrict__ block_descs,
 template <typename Backend, typename Impl, bool IsNoiseGen>
 template <typename T, typename Dist>
 void RNGBase<Backend, Impl, IsNoiseGen>::RunImplTyped(workspace_t<GPUBackend> &ws) {
-  using Block = BlockDesc<IsNoiseGen>;
   static_assert(std::is_same<Backend, GPUBackend>::value, "Unexpected backend");
   auto &output = ws.template OutputRef<GPUBackend>(0);
-  auto out_view = view<T>(output);
-  int nsamples = out_view.shape.size();
-  auto blocks_cpu = backend_data_.block_descs_cpu_.get();
-  auto blocks_gpu = backend_data_.block_descs_gpu_.get();
   auto rngs = backend_data_.randomizer_.states();
   int block_sz = backend_data_.block_size_;
   int max_nblocks = backend_data_.max_blocks_;
   int blockdesc_count = -1;
   TensorListView<StorageGPU, const T> in_view;
+  auto out_view = view<T>(output);
+  int nsamples = out_view.num_samples();
+  if (nsamples == 0) {
+    return;
+  }
+
   if (IsNoiseGen) {
     const auto& input = ws.template InputRef<GPUBackend>(0);
     in_view = view<const T>(input);
     output.SetLayout(input.GetLayout());
   }
-  blockdesc_count = SetupBlockDescs(
-    blocks_cpu, block_sz, max_nblocks, out_view, in_view);
+
+  // TODO(janton): set layout explicitly from the user for RNG
```

Comment:
This TODO is about random generators, not noise generators. We should do it at some point but not in this PR.